service/k8sclient/internal/kubectl/kubectl.go

```diff
@@ -60,11 +83,23 @@ func NewKubeCtl(ctx context.Context, kubeconfig string) (*KubeCtl, error) {
 		return nil, err
 	}
 
+	var cache *gocache.Cache
+	if len(options) == 1 && options[0]&UseCacheOption != 0 {
+		l.Info("kubectl cache is turned on")
+		// Setup cache to speed up long lasting requests.
+		// In some rpc methods of dbaas-controller, we call the same kubectl
+		// commands again but in different contexts. This is, I believe, most
+		// sane variant to share the data that were already requested. @jprukner
+		cache = gocache.New(expirationTime, expirationTime*2)
```

Comment:
since kubectl object is created for each request cache object is also is created for each request, using cache doesn't make sense. I checked gocache code and as far as I see there is no mechanisms to share cache between threads.

Reply:
Do we want to share it though? The idea was to cache in the scope of the request. Should I remove it entirely then?
Adding global cache would require map of caches actually, kubeconfig as key. We'd have to trim --kubeconfig out of the command tough because it's always different.

Reply:
cache removed, things we cached were moved from methods to their caller and reused.